content/faq-glossary-links/01-faq.en.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Frequently Asked Questions
+weight: 1
+---
+
+Some of the commonly asked questions encountered whilst running this course are addressed below, grouped into the main
+topics. If your questions is not answered here please do not hesitate to ask the course leaders or helpers.
+
+## Git
+
+### I got this error: "Push main: Push failed on refs/heads/main: push declined due to email privacy restrictions"
+
+A quick fix to this is to visit the [GitHub Email Settings](https://github.com/settings/emails) and uncheck `Block
+command line pushes that expose my email`. This is not ideal though as its exposes your email address. A better solution
+in the long term (described in detail on
+[StackOverflow](https://stackoverflow.com/questions/43378060/meaning-of-the-github-message-push-declined-due-to-email-privacy-restrictions))
+is to...
+
+1. visit the [GitHub Email Settings](https://github.com/settings/emails) and ensure `Keep my email address private` is
+   checked.
+2. Copy the `{ID}+{username}@users.noreply.github.com`.
+3. Open a [GitKraken CLI](https://www.gitkraken.com/cli#) window.
+4. Type `git config --global user.email {ID}+{username}@users.noreply.github.com` replacing `{ID}` and `{username}` with
+   the values you copied from above, then hit `Enter`.
+5. Amend the author of the previous commit with `git commit --ammend --reset-author`.
+6. Push the amended git with either `git push` at the Command Line Interface or using the `Push` button in GitKraken
+   Client.
+
+### Can you delete changes?
+
+You can but typically you make subsequent commits to revert changes. It is possible to roll back a branch to any stage
+in the history of commits by using `git checkout <hash_id>` but this will mean you are in a "Detached HEAD" state as you
+are not at the `HEAD` of that branch. It is possible to reset which commit `HEAD` points to but it is a destructive
```

Comment:
I don't really like this answer (of vague reset HEAD), there are multiple ways to delete commits depending on how/why you need to do it. Likewise, faq isn't really the place to discuss it as it requires a more thorough discussion (beyond just use `git revert`).

Reply:
I think that this is a tricky point, it's a question we're asked regularly (a FAQ, if you like) and I think it's worth having a  simple stock answer to it, even if it doesn't tell the whole story.

Verbally, I tend to stick with saying that it's possible to either add a new commit undoing what you have already committed, or to use revert (since this is easily available in gitkraken) as this answers the question fairly quickly, but without getting bogged down in all the options.

Perhaps we can flesh this section out a bit so we can refer people to simple and more complex versions of the answer in the FAQs, rather than try to explain it at all.